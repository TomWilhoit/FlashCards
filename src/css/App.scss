.App {
  text-align: center;
}

.review-header {
  display: flex;
  justify-content: center;
}

.question {
  display: flex;
  justify-content: center;
}
