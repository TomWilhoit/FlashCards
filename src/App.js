import React, { Component } from 'react';
import CardCountainer from './CardContainer';
import Timer from './Timer';
import Banner from './Banner';
import Counter from './Counter';
import Card from './Card';
import FlashCardDataSet from './FlashCardDataSet';
import './css/App.css';

class App extends Component {
  constructor() {
    super();


    this.state = {
      questions: FlashCardDataSet,
      shouldRestart: false,
      questionIndex: 0,
      savedArray: [],
    }
  }
 
  shouldRestart = () => {
    if (this.state.shouldRestart===false) {
    this.setState({
      shouldRestart: true
    })
  } else{
    this.setState({
      shouldRestart: false
      })
    }
  }

  incrementQuestionIndex = () => {
    const newIndex = this.state.questionIndex + 1;
    this.setState({
      questionIndex: newIndex
    })
  }

  saveToStorage = () => {
    let [...savedQuestions] = this.state.savedArray
    savedQuestions.push(this.state.questions[this.state.questionIndex])
    localStorage.setItem('savedQuestions',JSON.stringify(savedQuestions))
    this.setState({
      savedArray: savedQuestions
    })

      

    }
  
  
  
  render() {
    return (
      <div className="App">
        <Banner/>
        <CardCountainer
        questionObj= {this.state.questions[this.state.questionIndex]}
        incrementQuestionIndex = {this.incrementQuestionIndex}
        saveToStorage = {this.saveToStorage}/>
        <Timer shouldRestart={this.shouldRestart}/>
        <Counter
        questionIndex = {this.state.questionIndex}/>

      </div>
    ) };
  }

export default App;
