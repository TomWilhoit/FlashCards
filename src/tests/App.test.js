import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import {shallow} from 'enzyme'


describe('App', () => {
  let wrapper;
  



beforeEach(() => {
  localStorage.setItem('savedQuestions', '[{}, {}, {},]');
  wrapper = shallow(
    <App />
  )
});

afterEach(() => {
  localStorage.clear();
}); 


it('should update local storage', () => {
  let savedArray = [{}, {}, {}];
  wrapper.setState({ savedArray, questionIndex: 0 });
  expect(wrapper.state().savedArray.length).toEqual(3);
  wrapper.instance().saveToStorage();
  let savedQuestions = JSON.parse(localStorage.getItem('savedQuestions')).length;
  expect(savedQuestions).toEqual(4);
}); 

it('should have a proper default state', () => {
  expect(wrapper.state()).toEqual({
    questions: [],
    shouldRestart: false,
    questionIndex: 0,
    savedArray: [],
    wrongArray: []
  }); 
});

it('should match the snapshot with all data passed in correctly', () => {
  expect(wrapper).toMatchSnapshot();
});



// it('should pull from local storage', () => {
//   let returnedArray = wrapper.instance().pullFromStorage();
//   expect(returnedArray).toEqual('[{}, {}, {},]');
// }); 




// it('should change state when finish game is ran', () => {
//     wrapper.instance().finishGame = jest.fn();
//     wrapper.update();
//     expect(wrapper.instance().state()).toEqual();
//   }):




})