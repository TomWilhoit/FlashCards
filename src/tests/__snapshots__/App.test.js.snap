// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="App"
>
  <Banner />
  <CardContainer
    finishGame={[Function]}
    incrementQuestionIndex={[Function]}
    questionIndex={0}
    questions={Array []}
    saveToStorage={[Function]}
    shouldRestart={false}
  />
  <Timer
    shouldRestartGame={[Function]}
  />
  <Counter
    questionIndex={0}
  />
</div>
`;
